service/cvpartner.py

```diff
@@ -214,11 +200,51 @@ def __put_custom_tag(self, url, entity):
         logger.debug(entity["payload"])
         req = requests.put(url, headers=headers, json=entity["payload"])
         if req.status_code != 200:
-            logger.error(
-                "Unexpected response status code: %d with response text %s" % (req.status_code, req.text))
-            abort(req.status_code, req.json())
+            req = self.check_error(req, cv_url, headers, 'put', 'json', entity["payload"])
         return str(req.status_code)
 
+    def check_error(self, req, url, headers, method, data_type = None, data = None):
+        if req.status_code == 429:
+            return self.recursive_request(url, headers, req.headers.get('Retry-After'), method, data_type, data)                        
+        else:
+            logger.error("Unexpected response status code: %d with response text %s" % (req.status_code, req.text))
+            raise AssertionError("Unexpected response status code: %d with response text %s" % (req.status_code, req.text))        
+
+    def recursive_request(self, url, headers, retry_after, method, data_type = None, data = None):
```

Comment:
I suggest changing "data_type" to "json" to get rid of if-else on data_type.
Just suggest, though.